submissions/asaMitaka/FriendsApp/script.js

```diff
@@ -0,0 +1,237 @@
+    renderAllItemsToPage(friends);
+}
+
+form.gender.addEventListener('click', function(e) {
+    if (e.target && e.target.classList.contains('gender')) {
+        checkedGender(e);
+    }
+});
+
+function checkedGender(event) {
+    switch (event.target.value) {
+        case 'male':
+            renderMale();
+            break;
+        case 'female':
+            renderFemale();
+            break;
+        case 'all':
+            renderAll();
+            break;
+    }
+}
+
+function renderMale() {
+    friends = friendsCopy.slice();
+    friends = friends.filter(item=> item.gender === 'male');
+    renderAllItemsToPage(friends);
+}
+
+function renderFemale() {
+    friends = friendsCopy.slice();
+    friends = friends.filter(item=> item.gender === 'female');
+    renderAllItemsToPage(friends);
+}
+
+function renderAll() {
+    friends = friendsCopy.slice();
+    renderAllItemsToPage(friends);
+}
+
+form.minMax.addEventListener('change', checkNumber);    
+form.minMax.addEventListener('change', findByAge);
+
+function checkNumber() {
+    if (minNumber.value < 10) {
+        minNumber.value = 10
+        return;
+    } 
+    
+    if (maxNumber.value > 100 && minNumber.value < 10) {
+        maxNumber.value = 100;
+        minNumber.value = 10;
+        return;
+    } 
+    
+    if (maxNumber.value > 100) {
+        maxNumber.value = 100;
+        return;
+    } 
+
+    findByAge();
+}
+
+function findByAge() {
+    friends = friendsCopy.slice();
+
+    if (form.age.ageLess.checked) {
+        sortedAgeLess();
+    }
+
+    if (form.age.ageMore.checked) {
+        sortedAgeMore();
+    }
+
+    if (form.last.lastAToZ.checked) {
+        sortedAtoZLast();
+    }
+
+    if (form.last.lastZToA.checked) {
+        sortedZtoALast();
+    }
+
+    friends = friends.filter(item => {
+        if (item.dob.age > minNumber.value && item.dob.age <= maxNumber.value) {
+            return item;
+        }
+    });
+
+    renderAllItemsToPage(friends);
+}
+
+form.search.search.addEventListener('input', searchFunc);
+
+function searchFunc() {
+    isSearch = true;
+    let val = form.search.search.value.toLowerCase();
+    if (val && val.length > 0) {
+        content.innerHTML = '';
+
+        let acc = '';
+        friends.forEach(item=> {
+            if(item.name.first.toLowerCase().includes(val) || item.name.last.toLowerCase().includes(val)) {
+                acc += creatingDivItem(item);
+            }
+        });
+        content.innerHTML = acc;
+    } else {
+        isSearch = false;
+        renderAllItemsToPage(friends);
+    }
+}
+
+function resetPage() {
+    minNumber.value = 10;
+    maxNumber.value = 100;
+    friends = friendsCopy.slice();
+
+    renderAllItemsToPage(friends);
+}
+
+function renderAllItemsToPage(arr) {
+    if (form.search.search.value.length === 0) {
+        isSearch = false;
+    }
+
+    content.innerHTML = '';
+
+    let acc = '';
```

Comment:
Instead of creating a string of elements, it would be better to use the `map` method and create an array of elements, just join them before rendering. This would allow you to manipulate this array in the future if needed.